comp/logs/agent/config/endpoints.go

```diff
@@ -35,9 +38,19 @@ const (
 
 // Endpoint holds all the organization and network parameters to send logs to Datadog.
 type Endpoint struct {
-	apiKeyGetter func() string
-	isReliable   bool
-	useSSL       bool
+	isReliable bool
+	useSSL     bool
+
+	// the apiKey to use for this endpoint
+	apiKey *atomic.String
+	// The path of the config used to get the API key. This path is used to listen for configuration updates from
+	// the config.
+	configSettingPath string
+	// was this endpoint an "additional_endpoints"
+	isAdditionalEndpoint bool
```

Comment:
How about instead of keeping both `isAdditionalEndpoint` and `additionalEndpointsIdx`, we use a sentinel value like `-1` for the index of the main endpoint, and then get rid of this bool.

Possible downside with this approach: the zero value constructor no longer does the right thing. But I think that's okay as all the constructors should eventually either end up calling something in this file, or are covered by the other changes you're making in this PR.

Reply:
I think I prefer relying on the default values. Since `Endpoint` is a public type I want to avoid side effect bug if people start creating their own Endpoint without the constructor (the Agent code base being a bit chaotic I want to aim on the safe side).